### PR TITLE
djv: init at 2.0.8-unstable-2021-07-31

### LINKED ISSUE #68937: Package request: djv
**Project description**
Professional media review software for VFX, animation, and film production

**Metadata**
* homepage URL: http://djv.sourceforge.net/  
* source URL: https://github.com/darbyjohnston/DJV   
* license: BSD-like ([link](http://djv.sourceforge.net/Legal.html))  
* platforms:  
  * linux, http://djv.sourceforge.net/BuildLinux.html
  * darwin http://djv.sourceforge.net/BuildOSX.html

**FYI**
Arch AUR PKGBUILD (for reference)  
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=djv

### PR DESCRIPTION
**Pull request by blitz**
###### Motivation for this change

DJV provides professional review software for VFX, animation, and film production. 

I did package the master branch, because the last release is a bit older (2020) and the way third-party dependencies are built has changed somewhat since the 2.0.8 release.

This package was requested in #68937 by @luzpaz.

Fixes #68937.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/djv/default.nix [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+}:
+
+let
+
+  # The way third-party dependencies are packaged has changed
+  # significantly from the 2.0.8 release. This means any packaging
+  # effort for the 2.0.8 release would have to be redone for the next
+  # release. Hence we package the git version for now and can easily
+  # jump onto the next release once it's available.
+  djvVersion = "2.0.8-git20210731";
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/djv/default.nix:28]:
```suggestion
  djvVersion = "2.0.8-unstable=2021-07-31";
```

Just a bit of readability

File: pkgs/applications/graphics/djv/default.nix [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    rev = "ae31712c4f2802a874217ac194bde26287993934";
+    sha256 = "1qgia6vqb6fhyfj8w925xl6k6zidrp2gj5f32bpi94lwwhi6p9pd";
+  };
+
+  fseqSrc = fetchFromGitHub {
+    owner = "darbyjohnston";
+    repo = "fseq";
+
+    # DJV just pulls in the tip of the fseq master branch.
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/djv/default.nix:42]: I did not understand this comment.

Do DJV uses vendoring or something like?

**Review by tfc — APPROVED** [review]

File: pkgs/applications/graphics/djv/default.nix (hunk 2 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix:42]: The DJV build system pulls in some dependencies via CMake external projects functionality directly from Git.

File: pkgs/applications/graphics/djv/default.nix (hunk 1 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  # effort for the 2.0.8 release would have to be redone for the next
+  # release. Hence we package the git version for now and can easily
+  # jump onto the next release once it's available.
+  djvVersion = "2.0.8-unstable-2021-07-31";
+
+  djvSrc = fetchFromGitHub {
+    owner = "darbyjohnston";
+    repo = "djv";
+
+    rev = "ae31712c4f2802a874217ac194bde26287993934";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:34]:
```suggestion
    repo = "djv";
    rev = "ae31712c4f2802a874217ac194bde26287993934";
```

File: pkgs/applications/graphics/djv/default.nix [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  };
+
+  fseqSrc = fetchFromGitHub {
+    owner = "darbyjohnston";
+    repo = "fseq";
+
+    # DJV just pulls in the tip of the fseq master branch.
+    rev = "545fac6018100f7fca474b8ee4f1efa7cbf6bf45";
+
+    sha256 = "0qfhbrzji05hh5kwgd1wvq2lbf81ylbi7v7aqk28aws27f8d2hk0";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:45]:
```suggestion
    repo = "fseq";
    # DJV just pulls in the tip of the fseq master branch.
    rev = "545fac6018100f7fca474b8ee4f1efa7cbf6bf45";
    sha256 = "0qfhbrzji05hh5kwgd1wvq2lbf81ylbi7v7aqk28aws27f8d2hk0";
```

File: pkgs/applications/graphics/djv/default.nix [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    sourceRoot = "source/etc/SuperBuild";
+
+    nativeBuildInputs = [ cmake ];
+    buildInputs = [
+      libGL
+    ];
+
+    postPatch = ''
+      # Our source directory is read-only. This prevents patching.
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:62]:
```suggestion
```

File: pkgs/applications/graphics/djv/default.nix [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+      sed -i '/list(APPEND DJV_THIRD_PARTY_DEPS GLFW)/d' \
+          CMakeLists.txt
+      sed -i '/list(APPEND DJV_THIRD_PARTY_DEPS ZLIB)/d' \
+          CMakeLists.txt
+      sed -i '/list(APPEND DJV_THIRD_PARTY_DEPS PNG)/d' \
+          CMakeLists.txt
+      sed -i '/list(APPEND DJV_THIRD_PARTY_DEPS FreeType)/d' \
+          CMakeLists.txt
+      sed -i '/list(APPEND DJV_THIRD_PARTY_DEPS OCIO)/d' \
+          CMakeLists.txt
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:88]:
```suggestion
      sed -i CMakeLists.txt \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS RapidJSON)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS RtAudio)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS IlmBase)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS GLM)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS GLFW)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS ZLIB)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS PNG)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS FreeType)/d' \
        -e '/list(APPEND DJV_THIRD_PARTY_DEPS OCIO)/d'
```

File: pkgs/applications/graphics/djv/default.nix [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    cmakeFlags = [
+      "-DDJV_THIRD_PARTY_OpenEXR:BOOL=False"
+      "-DDJV_THIRD_PARTY_JPEG:BOOL=False"
+      "-DDJV_THIRD_PARTY_TIFF:BOOL=False"
+    ];
+
+    installPhase = ''
+      echo Already installed.
+    '';
+
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:108]:
```suggestion
    dontInstall = true;
```

File: pkgs/applications/graphics/djv/default.nix [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    glfw3
+    zlib.dev
+    libpng
+    freetype
+    opencolorio_1
+    djv-deps
+  ];
+
+  postPatch = ''
+    # Our source directory is read-only. This prevents patching.
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:139]:
```suggestion
```

File: pkgs/applications/graphics/djv/default.nix [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+    # When linking opencolorio statically this results in failing to
+    # pull in opencolorio's dependencies (tixml and yaml libraries). Avoid
+    # this by linking it statically instead.
+
+    sed -i 's/PATH_SUFFIXES static//' \
+        cmake/Modules/FindOCIO.cmake
+
+    sed -i '/OpenColorIO_STATIC/d' \
+        cmake/Modules/FindOCIO.cmake
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:150]:
```suggestion
    sed cmake/Modules/FindOCIO.cmake -i \
      -e 's/PATH_SUFFIXES static//' \
      -e '/OpenColorIO_STATIC/d'
```

File: pkgs/applications/graphics/djv/default.nix [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+    # DJV also supports MacOS, but this needs someone with a Mac to
+    # test it.
+    platforms = platforms.linux;
+
+    maintainers = [ maintainers.blitz ];
+    license = licenses.bsd3;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:168]:
```suggestion
    homepage = "https://darbyjohnston.github.io/DJV/";
    platforms = platforms.linux;
    maintainers = [ maintainers.blitz ];
    license = licenses.bsd3;
  };
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/djv/default.nix (hunk 2 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: I've tried to clarify the comment.

File: pkgs/applications/graphics/djv/default.nix (hunk 3 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 4 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 5 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 6 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 7 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 8 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 9 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

File: pkgs/applications/graphics/djv/default.nix (hunk 10 above)
Comment (blitz) [reply pkgs/applications/graphics/djv/default.nix]: Applied.

**Comment by blitz** [comment]
@SuperSandro2000 Thanks for the review! I've removed the pointless newlines and simplified the sed expressions. Please take a look again.

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
@blitz 

A last thing: the commit message

> Fixes NixOS#68937

is not necessary, and can be confusing. 

The `#` numbers are Github-specific, and it is not a good practice to put this info. If, e.g., we are faced with the necessity of a migration to a new server, a message like "Fixes #68937" becomes meaningless.

Just remove it.

**Comment by blitz** [comment]
> @blitz
> 
> A last thing: the commit message
> 
> > Fixes #68937
> 
> is not necessary, and can be confusing.
> 
> The `#` numbers are Github-specific, and it is not a good practice to put this info. If, e.g., we are faced with the necessity of a migration to a new server, a message like "Fixes #68937" becomes meaningless.
> 
> Just remove it.

Good point. I've removed it.

File: pkgs/applications/graphics/djv/default.nix [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    libGL
+    libX11
+    libXinerama
+    libXi
+    rapidjson
+    rtaudio
+    ilmbase
+    glm
+    glfw3
+    zlib.dev
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/djv/default.nix:123]: .dev is unecessary.

**Pull request merged**